src/examples/3d_fluid_sim/fluid_sim.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#ifndef DIM
+#define DIM 3
+#endif
+
+#if DIM == 2
+#define DIM 3
+#endif
+
+#include <random>
```

Comment:
Minor: I personally put all `include`s before `define`s

Reply:
It was done this way to avoid getting the default initialisation of DIM to be 2 from the vector (I think) header file